src/modules/status-list/status-list.service.ts

```diff
@@ -0,0 +1,113 @@
+import { ConflictException, Injectable } from '@nestjs/common';
+import StatusList from '@digitalbazaar/vc-status-list';
+import { URL } from 'url';
+import { ConfigService } from '@nestjs/config';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import {
+  CredentialWithStatusDto,
+  CredentialStatusDto,
+} from './dtos/credential-status.dto';
+import { STATUS_LIST_MODULE_PATH } from './status-list.const';
+import { CredentialDto } from './dtos/credential.dto';
+import { VerifiableCredentialDto } from './dtos/verifiable-credential.dto';
+import { StatusListCredentialDto } from './dtos/status-list-credential.dto';
+import { StatusListVerifiableCredentialDto } from './dtos/status-list-verifiable-credential.dto';
+import { StatusListCredential } from './entities/status-list-credential.entity';
+
+@Injectable()
+export class StatusListService {
+  constructor(
+    private readonly configService: ConfigService,
+    @InjectRepository(StatusListCredential)
+    private readonly statusListRepository: Repository<StatusListCredential>
+  ) {}
+
+  async createEntry(
+    credential: CredentialDto
+  ): Promise<CredentialWithStatusDto> {
+    const statusListCredential = new URL(
+      `${STATUS_LIST_MODULE_PATH}/${credential.id}`,
+      this.configService.get('STATUS_LIST_DOMAIN')
+    );
+
+    const credentialStatus: CredentialStatusDto = {
+      type: 'StatusList2021Entry',
+      statusPurpose: 'revocation',
+      statusListIndex: '1',
+      statusListCredential: statusListCredential.href,
+    };
+
+    return { ...credential, credentialStatus };
+  }
+
+  async createStatusListCredential(
+    credential: VerifiableCredentialDto,
+    currentUser: string
+  ): Promise<StatusListCredentialDto> {
+    const statusList = await this.statusListRepository.findOne({
+      where: {
+        credentialId: credential.id,
+      },
+    });
+
+    if (statusList) {
+      throw new ConflictException('Credential already revoked');
+    }
+
+    const list = await StatusList.createList({ length: 16000 });
```

Comment:
Why length `16000` here? I think we only need length `1` because we going to do a 1 to 1 mapping from VC to StatusList2021Credential
```suggestion
    const list = await StatusList.createList({ length: 1 });
```

Reply:
> The encodedList property of the credential [subject](https://w3c-ccg.github.io/vc-status-list-2021/#dfn-subjects) MUST be the GZIP-compressed [[RFC1952](https://w3c-ccg.github.io/vc-status-list-2021/#bib-rfc1952)], base-64 encoded [[RFC4648](https://w3c-ccg.github.io/vc-status-list-2021/#bib-rfc4648)] bitstring values for the associated range of [verifiable credential](https://w3c-ccg.github.io/vc-status-list-2021/#dfn-verifiable-credentials) status values. The uncompressed bitstring MUST be at least **16KB** in size.

see : https://w3c-ccg.github.io/vc-status-list-2021/#statuslist2021credential

Reply:
Hmm, I think this 16KB minimum is for privacy reasons that don't really apply in our case. Or can you think of a good reason for the 16KB minimum?

Reply:
I think just the privacy reasons.